### PR TITLE
AF-1080: New project with the same name as another project in different space disappear after server restarts

### PR DESCRIPTION
**Pull request by paulovmr**
Part of an ensemble:
* https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/723
* https://github.com/kiegroup/appformer/pull/251
* https://github.com/kiegroup/kie-wb-common/pull/1520

=============================================================

Old system.git structure example:
```
.
├── anotherOU-myrepo.repository
├── anotherOU.organizationalunit
├── another-repo.repository
├── myrepo.repository
├── myteam.organizationalunit
├── readme.md
├── settings.global
└── work-items-editor-settings.editor
```

New system.git structure example:
```
.
├── deployments
├── editors
│   └── work-items-editor-settings.editor
├── global
│   └── settings.global
├── projects
├── readme.md
├── repositories
│   ├── anotherOU
│   │   ├── project1-inside-anotherOU-myrepo.repository
│   │   └── project2-inside-anotherOU-myrepo.repository
│   └── myteam
│       ├── project-inside-another-repo.repository
│       └── project-inside-myrepo.repository
└── spaces
    ├── anotherOU.space
    └── myteam.space
```

### CODE REVIEW

**Comment by paulovmr** [comment]
jenkins execute full downstream build

File: kie-wb-common-cli/kie-wb-common-cli-tools/kie-wb-common-cli-forms-migration/src/main/java/org/kie/workbench/common/forms/migration/tool/FormsMigrationTool.java [hunk 1]
```diff
@@ -170,10 +176,20 @@ public FileVisitResult visitFile(org.uberfire.java.nio.file.Path visitedPath, Ba
         pipeline.migrate(context);
     }
 
-    private boolean validateTarget() {
-
+    private boolean projectMigrationWasExecuted() {
         if (!config.getTarget().resolve("system").resolve(MigrationConstants.SYSTEM_GIT).toFile().exists()) {
-            system.err().println(String.format("The target path (%s) looks like it doesn't contain an updated filesystem, %s must be excuted before migrating forms", niogitDir, ProjectMigrationTool.NAME.toUpperCase()));
+            system.err().println(String.format("The PROJECT STRUCTURE MIGRATION must be ran before this one."));
```
Comment (barboras7) [inline kie-wb-common-cli/kie-wb-common-cli-tools/kie-wb-common-cli-forms-migration/src/main/java/org/kie/workbench/common/forms/migration/tool/FormsMigrationTool.java]: Redundant call to String.format(), in this case it's not necessary as there are no arguments provided.

File: kie-wb-common-cli/kie-wb-common-cli-tools/kie-wb-common-cli-project-migration/src/main/java/org/kie/workbench/common/project/config/MigrationConfigurationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+                                                                                entry -> {
+                                                                                    if (!Files.isDirectory(entry) &&
+                                                                                            !entry.getFileName().toString().startsWith(".") &&
+                                                                                            entry.getFileName().toString().endsWith(type.getExt())) {
+                                                                                        return true;
+                                                                                    }
+                                                                                    return false;
+                                                                                }
+        );
+
```
Comment (barboras7) [inline kie-wb-common-cli/kie-wb-common-cli-tools/kie-wb-common-cli-project-migration/src/main/java/org/kie/workbench/common/project/config/MigrationConfigurationServiceImpl.java]: No need to use the "if" statement, just call return on the condition.

File: kie-wb-common-cli/kie-wb-common-cli-tools/kie-wb-common-cli-system-configuration-migration/src/main/java/org/kie/workbench/common/system/configuration/ConfigGroupsMigrationService.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        return ioService.newDirectoryStream(dir,
+                                            entry -> {
+                                                if (!Files.isDirectory(entry) &&
+                                                        !entry.getFileName().toString().startsWith(".") &&
+                                                        entry.getFileName().toString().endsWith(extension)) {
+                                                    return true;
+                                                }
+                                                return false;
+                                            });
+    }
```
Comment (barboras7) [inline kie-wb-common-cli/kie-wb-common-cli-tools/kie-wb-common-cli-system-configuration-migration/src/main/java/org/kie/workbench/common/system/configuration/ConfigGroupsMigrationService.java]: Same as above mentioned "if" statement.

File: kie-wb-common-cli/kie-wb-common-cli-tools/kie-wb-common-cli-system-configuration-migration/src/main/java/org/kie/workbench/common/system/configuration/SystemConfigurationMigrationTool.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        } catch (Throwable ignore) {
+            // Suppress exceptions from bad shutdown
+        }
+    }
+
+    private boolean validateTarget() {
+        if (!config.getTarget().resolve("system").resolve(MigrationConstants.SYSTEM_GIT).toFile().exists()) {
+            system.err().println(String.format("The PROJECT STRUCTURE MIGRATION must be ran before this one."));
+            return false;
+        }
```
Comment (barboras7) [inline kie-wb-common-cli/kie-wb-common-cli-tools/kie-wb-common-cli-system-configuration-migration/src/main/java/org/kie/workbench/common/system/configuration/SystemConfigurationMigrationTool.java]: Redundant call to String.format().

File: kie-wb-common-cli/kie-wb-common-cli-tools/kie-wb-common-cli-project-migration/src/main/java/org/kie/workbench/common/project/config/MigrationConfigurationServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+        final List<ConfigGroup> configGroups = new ArrayList<>();
+        final DirectoryStream<Path> foundConfigs = ioService.newDirectoryStream(ioService.get(systemRepository.getUri()),
+                                                                                entry -> {
+                                                                                    if (!Files.isDirectory(entry) &&
+                                                                                            !entry.getFileName().toString().startsWith(".") &&
+                                                                                            entry.getFileName().toString().endsWith(type.getExt())) {
+                                                                                        return true;
+                                                                                    }
+                                                                                    return false;
```
Comment (barboras7) [inline kie-wb-common-cli/kie-wb-common-cli-tools/kie-wb-common-cli-project-migration/src/main/java/org/kie/workbench/common/project/config/MigrationConfigurationServiceImpl.java]: No need to use the "if" statement, just call return on the condition.

**Pull request merged**